src/gc/gcenv.windows.cpp

```diff
@@ -0,0 +1,697 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#include <cstdint>
+#include <cassert>
+#include <cstddef>
+#include <memory>
+#include "windows.h"
+#include "psapi.h"
+#include "env/gcenv.structs.h"
+#include "env/gcenv.base.h"
+#include "env/gcenv.os.h"
+
+#ifndef FEATURE_STANDALONE_GC
+#error This file should only be compiled for a standalone GC
+#endif // FEATURE_STANDALONE_GC
+
+GCSystemInfo g_SystemInfo;
+
+typedef BOOL (WINAPI *PGET_PROCESS_MEMORY_INFO)(HANDLE handle, PROCESS_MEMORY_COUNTERS* memCounters, uint32_t cb);
+static PGET_PROCESS_MEMORY_INFO GCGetProcessMemoryInfo = 0;
+
+static size_t g_RestrictedPhysicalMemoryLimit = (size_t)UINTPTR_MAX;
+
+typedef BOOL (WINAPI *PIS_PROCESS_IN_JOB)(HANDLE processHandle, HANDLE jobHandle, BOOL* result);
+typedef BOOL (WINAPI *PQUERY_INFORMATION_JOB_OBJECT)(HANDLE jobHandle, JOBOBJECTINFOCLASS jobObjectInfoClass, void* lpJobObjectInfo, DWORD cbJobObjectInfoLength, LPDWORD lpReturnLength);
+
+namespace {
+
+void GetProcessMemoryLoad(LPMEMORYSTATUSEX pMSEX)
+{
+    pMSEX->dwLength = sizeof(MEMORYSTATUSEX);
+    BOOL fRet = ::GlobalMemoryStatusEx(pMSEX);
+    assert(fRet);
+
+    // If the machine has more RAM than virtual address limit, let us cap it.
+    // Our GC can never use more than virtual address limit.
+    if (pMSEX->ullAvailPhys > pMSEX->ullTotalVirtual)
+    {
+        pMSEX->ullAvailPhys = pMSEX->ullAvailVirtual;
+    }
+}
+
+#ifdef FEATURE_CORECLR
+
+inline bool RunningOnWin8()
+{
+    // TODO(segilles) platform detection
+    return false;
+}
+
+// For coresys we need to look for an API in some apiset dll on win8 if we can't find it  
+// in the traditional dll.
+HINSTANCE LoadDllForAPI(const WCHAR* dllTraditional, const WCHAR* dllApiSet)
+{
+    HINSTANCE hinst = LoadLibraryEx(dllTraditional, nullptr, 0);
+
+    if (!hinst)
+    {
+        if(RunningOnWin8())
+            hinst = LoadLibraryEx(dllApiSet, nullptr, 0);
+    }
+
+    return hinst;
+}
+#endif
+
+static size_t GetRestrictedPhysicalMemoryLimit()
+{
+    LIMITED_METHOD_CONTRACT;
+
+    // The limit was cached already
+    if (g_RestrictedPhysicalMemoryLimit != (size_t)UINTPTR_MAX)
+        return g_RestrictedPhysicalMemoryLimit;
+
+    size_t job_physical_memory_limit = (size_t)UINTPTR_MAX;
+    BOOL in_job_p = FALSE;
+#ifdef FEATURE_CORECLR
```

Comment:
This should not be under `FEATURE_CORECLR` ifdef - the CoreCLR path should be used all the time.